client/consensus/qpow/src/lib.rs

```diff
@@ -92,22 +65,27 @@ where
 		};
 		let parent_hash = match extract_block_hash(parent) {
 			Ok(hash) => hash,
-			Err(_) => return Ok(false),
+			Err(_) => return Ok((false, U512::zero(), U512::zero())),
 		};
 
 		let pre_hash = pre_hash.as_ref().try_into().unwrap_or([0u8; 32]);
-
-		// Verify the nonce using QPoW
-		if !self
+		let (verified, difficulty, distance_achieved) = self
 			.client
 			.runtime_api()
-			.verify_for_import(parent_hash, pre_hash, nonce)
-			.map_err(|e| Error::Runtime(format!("API error in verify_nonce: {:?}", e)))?
-		{
-			return Ok(false);
+			.verify_current_block(parent_hash, pre_hash, nonce, false)
+			.map_err(|e| Error::Runtime(format!("API error in verify_nonce: {:?}", e)))?;
+
+		// Verify the nonce using QPoW
```

Comment:
this comment may be out of place - i am happy to delete most comments..

Reply:
yeah it would be more appropriate to put it on top of the verify_current_block call, not below

Reply:
also not sure why _difficulty is even an argument here, the verify call pulls it from the runtime...ugh ok i will clean this up